sycl/test/hier_par/hier_par_wgscope.cpp

```diff
@@ -6,279 +6,18 @@
-  return false;
-}
-
-template <typename GoldFnTy>
-bool verify(int testcase, int RangeLength, int *Ptr, GoldFnTy get_gold) {
-  int ErrCnt = 0;
-
-  for (int I = 0; I < RangeLength; I++) {
-    int Gold = get_gold(I);
-
-    if (Ptr[I] != Gold) {
-      if (++ErrCnt < 20) {
-        std::cout << testcase << " - ERROR at " << I << ": " << Ptr[I]
-                  << " != " << Gold << "(expected)\n";
-      }
-    }
-  }
-  if (ErrCnt > 0)
-    std::cout << "-- Failure rate: " << ErrCnt << "/" << RangeLength << "("
-              << ErrCnt / (float)RangeLength * 100.f << "%)\n";
-  return ErrCnt == 0;
-}
-
-struct MyStruct {
-  size_t x;
-  size_t y;
-};
-
-// Tests complex private_memory usage:
-// - Data type is a structure
-// - two different variables are used, both are live across two PFWI scopes
-bool testPrivateMemory(queue &Q) {
-  std::cout << "+++ Testing private_memory class implementation...\n";
-  constexpr int N_ITER = 3;
-  constexpr int N_WG = 2;
-  constexpr int WG_X_SIZE = 7;
-  constexpr int WG_Y_SIZE = 3;
-  constexpr int WG_LINEAR_SIZE = WG_X_SIZE * WG_Y_SIZE;
-  constexpr int RangeLength = N_WG * WG_LINEAR_SIZE;
-  constexpr int C1 = 5;
-  constexpr int C2 = 1;
-
-  std::unique_ptr<int[]> Data(new int[RangeLength]);
-  int *Ptr = Data.get();
-
-  std::memset(Ptr, 0, RangeLength * sizeof(Ptr[0]));
-  buffer<int, 1> Buf(Ptr, range<1>(RangeLength));
-  Q.submit([&](handler &cgh) {
-    auto DevPtr = Buf.get_access<access::mode::read_write>(cgh);
-
-    cgh.parallel_for_work_group<class hpar_priv_mem>(
-        range<2>(N_WG, 1), range<2>(WG_X_SIZE, WG_Y_SIZE), [=](group<2> G) {
-          private_memory<MyStruct, 2> Priv1(G);
-          private_memory<MyStruct, 2> Priv2(G);
-
-          for (int Cnt = 0; Cnt < N_ITER; Cnt++) {
-            G.parallel_for_work_item(
-                range<2>(WG_X_SIZE, WG_Y_SIZE), [&](h_item<2> I) {
-                  auto GlobId = I.get_global().get_linear_id();
-                  DevPtr[GlobId]++;
-                  MyStruct &S1 = Priv1(I);
-                  S1.x = GlobId;
-                  S1.y = C1;
-                  MyStruct &S2 = Priv2(I);
-                  S2.x = C2;
-                  S2.y = GlobId;
-                });
-            G.parallel_for_work_item(range<2>(WG_X_SIZE, WG_Y_SIZE),
-                                     [&](h_item<2> I) {
-                                       MyStruct &S1 = Priv1(I);
-                                       MyStruct &S2 = Priv2(I);
-                                       DevPtr[I.get_global().get_linear_id()] +=
-                                           S1.x + S1.y + S2.x + S2.y;
-                                     });
-          }
-        });
-  });
-  auto Ptr1 = Buf.get_access<access::mode::read>().get_pointer();
-  bool Res = verify(0, RangeLength, Ptr1, [&](int I) -> int {
-    return N_ITER * (1 + C1 + C2 + 2 * I);
-  });
-  std::cout << (Res ? "  Passed\n" : "  FAILED\n");
-  return Res;
-}
+#include "hier_par_wgscope_impl.src"
```

Comment:
I suggest move this file to "Inputs" directory and change the file extension to `hpp`.
```suggestion
#include "Inputs/hier_par_wgscope_impl.hpp"
```